src/main/java/Application.java

```diff
@@ -0,0 +1,45 @@
+import java.util.Arrays;
+import java.util.Optional;
+import java.util.Scanner;
+
+public class Application {
+    public static void main(String[] args) {
+        Scanner scanner =  new Scanner(System.in);
+        System.out.println("경주할 자동차 이름을 입력하세요(이름은 쉼표(,)를 기준으로 구분");
+        String[] racingCarName = scanner.nextLine().split(",");
+
+        if (racingCarName.length > 5) {
+            System.out.println("자동차는 최대 5대까지만 입력할 수 있습니다.");
+            return;
+        }
+
+        Optional<String> invalidName = Arrays.stream(racingCarName)
+                .map(String::trim)
+                .filter(name -> name.length() > 5)
+                .findFirst();
+
+        if (invalidName.isPresent()) {
+            System.out.println("자동차 이름은 5자 이하여야 합니다: " + invalidName.get());
+            return;
+        }
```

Comment:
예외로 처리한다면 다음과 같은 방식으로도 처리가 가능해보이네요!

```suggestion
        Optional<String> invalidName = Arrays.stream(racingCarName)
                .map(String::trim)
                .filter(name -> name.length() > 5)
                .findFirst()
                orElseThrow(() -> new IllegalArgumentException("자동차 이름은 5자 이하여야 합니다"));

        if (invalidName.isPresent()) {
            System.out.println("자동차 이름은 5자 이하여야 합니다: " + invalidName.get());
            return;
        }
```

+) 이와  별개로 이름을 검증하는 책임은 Car 객체의 책임이 아닐까? 생각이 드는데 지수님은 어떻게 생각하시나요?

Reply:
예외 처리를 해 자동차 이름을 잘 못 입력한다면 메세지와 함께 프로그램을 종료하는 방법도 좋은 생각이네요!